components/BottomSheet/tests/unit/BottomSheetShapeThemerTests.swift

```diff
@@ -0,0 +1,70 @@
+// Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import XCTest
+import MaterialComponents.MaterialBottomSheet
+import MaterialComponents.MaterialBottomSheet_ShapeThemer
+
+class BottomSheetShapeThemerTests: XCTestCase {
+
+  func testBottomSheetShapeThemerExtended() {
+    // Given
+    let shapeScheme = MDCShapeScheme()
+    let bottomSheet = MDCBottomSheetController(contentViewController: UIViewController())
+    shapeScheme.largeSurfaceShape = MDCShapeCategory(cornersWith: .angled, andSize: 10)
+    bottomSheet.setShapeGenerator(MDCRectangleShapeGenerator(), for: .extended)
+
+    // When
+    MDCBottomSheetControllerShapeThemer.applyShapeScheme(shapeScheme, to: bottomSheet)
+
+    // Then
+    let extendedShapeGenerator = bottomSheet.shapeGenerator(for: .extended)
+    XCTAssert(extendedShapeGenerator is MDCRectangleShapeGenerator);
+    XCTAssertEqual((extendedShapeGenerator as! MDCRectangleShapeGenerator).topLeftCorner,
+                   shapeScheme.largeSurfaceShape.topLeftCorner.cornerTreatmentValue())
+    XCTAssertEqual((extendedShapeGenerator as! MDCRectangleShapeGenerator).topRightCorner,
+                   shapeScheme.largeSurfaceShape.topRightCorner.cornerTreatmentValue())
+    XCTAssertEqual((extendedShapeGenerator as! MDCRectangleShapeGenerator).bottomLeftCorner,
+                   shapeScheme.largeSurfaceShape.bottomLeftCorner.cornerTreatmentValue())
+    XCTAssertEqual((extendedShapeGenerator as! MDCRectangleShapeGenerator).bottomRightCorner,
+                   shapeScheme.largeSurfaceShape.bottomRightCorner.cornerTreatmentValue())
+  }
+
+  func testBottomSheetShapeThemerPreferred() {
+    // Given
+    let bottomSheetCollapsedBaselineShapeValue = NSNumber(value: 24)
```

Comment:
That we need to create an NSNumber in order to initialize this object is concerning. I've filed https://github.com/material-components/material-components-ios/issues/5063 to look into this - please do so following this change before too much code relies on this.

Reply:
ok.